main/exercise/question_ptest_admin.inc.php

```diff
@@ -0,0 +1,91 @@
+    $action = api_get_self().'?exerciseId='.$objExercise->id.'&'.api_get_cidreq().'&modifyQuestion='.$modifyQuestion.'&editQuestion='.$objQuestion->id.'&page='.$page;
+} else {
+    $objQuestion = Question::getInstance($_REQUEST['answerType'], true);
+    $action = api_get_self().'?exerciseId='.$objExercise->id.'&'.api_get_cidreq().'&modifyQuestion='.$modifyQuestion.'&newQuestion='.$newQuestion;
+}
+
+if (is_object($objQuestion)) {
+    // FORM CREATION
+    $form = new FormValidator('question_admin_form', 'post', $action);
+    if (isset($_GET['editQuestion'])) {
+        $class = 'btn btn-default';
+        $text = get_lang('ModifyQuestion');
+        $type = isset($_GET['type']) ? Security::remove_XSS($_GET['type']) : null;
+    } else {
+        $class = 'btn btn-default';
+        $text = get_lang('AddQuestionToExercise');
+        $type = $_REQUEST['answerType'];
+    }
+
+    $typesInformation = Question::getQuestionPtTypeList();
+    $form_title_extra = isset($typesInformation[$type][1]) ? get_lang($typesInformation[$type][1]) : null;
+
+    $code = '';
+    if (isset($objQuestion->code) && !empty($objQuestion->code)) {
+        $code = ' ('.$objQuestion->code.')';
+    }
+
+    // form title
+    $form->addHeader($text.': '.$form_title_extra.$code);
+
+    // question form elements
+    $objQuestion->createPtForm($form, $objExercise);
+
+    // answer form elements
+    $objQuestion->createAnswersForm($form);
+
+    // this variable  $show_quiz_edition comes from admin.php blocks the exercise/quiz modifications
+    if (!empty($objExercise->id) && $objExercise->edit_exercise_in_lp == false) {
+        $form->freeze();
+    }
+
+    // FORM VALIDATION
+    if (isset($_POST['submitQuestion']) && $form->validate()) {
+        // Question
+        $objQuestion->processCreation($form, $objExercise);
+        $objQuestion->processAnswersCreation($form, $objExercise);
+
+        if (isset($_GET['editQuestion'])) {
+            if (empty($exerciseId)) {
+                Display::addFlash(Display::return_message(get_lang('ItemUpdated')));
+                $url = 'ptest_admin.php?exerciseId='.$exerciseId.'&'.api_get_cidreq().'&editQuestion='.$objQuestion->id;
+                echo '<script type="text/javascript">window.location.href="'.$url.'"</script>';
+                exit;
+            }
+            echo '<script type="text/javascript">window.location.href="ptest_admin.php?exerciseId='.$exerciseId.'&'.api_get_cidreq().'&page='.$page.'&message=ItemUpdated"</script>';
+        } else {
+            // New question
+            $page = 1;
+            $length = api_get_configuration_value('question_pagination_length');
+            if (!empty($length)) {
+                $page = round($objExercise->getQuestionCount() / $length);
+            }
+            echo '<script type="text/javascript">window.location.href="ptest_admin.php?exerciseId='.$exerciseId.'&'.api_get_cidreq().'&page='.$page.'&message=ItemAdded"</script>';
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line exceeds maximum limit of 120 characters; contains 179 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5018478)